### ISSUE TITLE
Code Design for intermediate processing steps requiring external or dependent parameters

### ISSUE THREAD

**Issue by Sohambutala** [OSOceanAcoustics/echoflow#25]
I was searching for sample use cases for applying masks in echoflow. Specifically, we're utilizing the frequency_differencing function and employing it as a mask. The challenge arises from our utilization of lists for freqAB and chanAB, and echoflow using YAML configuration. The question is: how should I accommodate the input, considering that these lists are generated from the "sv" output?

Here's an example of how we're currently approaching this:

`freqAB = list(out_ds.frequency_nominal.values[:2])
        freqdiff_da = ep.mask.frequency_differencing(source_Sv=out_ds, freqAB=freqAB, operator=">", diff=5)`

This is merely a singular instance; there could exist additional functions that also rely on preceding data. An illustration of this is the `add_location` function, which requires both an xarray dataset and an echodata object to perform a location update.

@leewujung @lsetiawan @valentina-s

**Comment by Sohambutala** [comment]
Passing a file after existence check wherever a file is passed.

Example:

```{yaml}
- name: echoflow_add_location
    module: echoflow.stages.subflows.add_location
    options:
      use_offline: true
    external_params:
      echodata: 'path/to/file'
```

**Comment by Sohambutala** [comment]
Closed in #45
